### PR TITLE
Update some features

### PR DESCRIPTION
**Pull request by asdflj**
Update some features
1.click record to rename P2P Tunnel (Help player to build large ae network)
2.shift click record. Well be transport player to P2P Tunnel position (OP only)
image
image

### CODE REVIEW

File: src/main/java/com/projecturanus/betterp2p/client/gui/GuiRenameItem.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.projecturanus.betterp2p.client.gui
+
+import appeng.client.gui.widgets.MEGuiTextField
+import com.projecturanus.betterp2p.MODID
+import com.projecturanus.betterp2p.client.TextureBound
+import net.minecraft.client.gui.GuiScreen
+import net.minecraft.util.ResourceLocation
+
+class GuiRenameItem(msg: InfoWrapper): GuiScreen(), TextureBound {
```
Comment (GlodBlock) [inline src/main/java/com/projecturanus/betterp2p/client/gui/GuiRenameItem.kt:9]: what this gui for?

File: src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetP2PDevice.kt [hunk 2]
```diff
@@ -82,6 +87,10 @@ class WidgetP2PDevice(private val selectedInfoProperty: KMutableProperty0<InfoWr
     }
 
     private fun drawButtons(gui: GuiScreen, info: InfoWrapper, mouseX: Int, mouseY: Int, partialTicks: Float) {
+        info.renameButton.xPosition = x
```
Comment (GlodBlock) [inline src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetP2PDevice.kt:90]: why the rename button is the whole widget

File: src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    for (part in listP2P!!){
+        var a =part.side.getEnumIndex()
+        if(part.side.getEnumIndex() == message.info.side){ return part}
+    }
+    return null
+}
+
+class ServerRenameP2PTunnel : IMessageHandler<C2SP2PTunnelInfo, IMessage?> {
+    override fun onMessage(message: C2SP2PTunnelInfo, ctx: MessageContext): IMessage? {
+        val world: World =  DimensionManager.getWorld(message.info.world)
```
Comment (GlodBlock) [inline src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:25]: i think you should also send the `S2CRefreshInfo` back to client side, so rename change can be updated in gui

File: src/main/java/com/projecturanus/betterp2p/client/gui/GuiRenameItem.kt (hunk 1 above)
Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/client/gui/GuiRenameItem.kt:9]: This is automatically generated. I forget delete it

File: src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetP2PDevice.kt (hunk 2 above)
Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetP2PDevice.kt:90]: eeeeeeeeeee

File: src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt (hunk 3 above)
Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:25]: send name before is already updated name in the gui

File: src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetP2PDevice.kt (hunk 2 above)
Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetP2PDevice.kt:90]: Y4P _Z694VZ37848)F FWYX

File: src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt (hunk 3 above)
Comment (GlodBlock) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:25]: image
the p2ps with the same frq do not change their name after renaming

Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:25]: eeeeeeee  I think there is something wrong with my solution,  I will close this pr

**Pull request closed without merging**